data/lib/src/network/linshare_http_client.dart

```diff
@@ -734,11 +737,32 @@ class LinShareHttpClient {
   }
 
   Future<String> getOIDCConfiguration(Uri baseUrl) async {
-    final result = await _dioClient.get(
-        Endpoint.oidcConfiguration.generateOIDCConfigurationUrl(baseUrl));
+    final result = await _dioClient.get(Endpoint.oidcConfiguration.generateBaseUrl(baseUrl));
     return result;
   }
 
+  Future<SecretTokenResponse> getSaaSSecretToken(Uri baseUrl, PlanBodyRequest planBodyRequest) async {
+    final result = await _dioClient.post(
+        Endpoint.secretToken.generateBaseUrl(baseUrl),
+        data: planBodyRequest.toJson());
+    return SecretTokenResponse.fromJson(result);
+  }
+
+  Future<VerifyEmailResponse> verifyEmailSaaS(Uri baseUrl, String email) async {
+    final result = await _dioClient.get(
+        Endpoint.verifyEmail
+            .withPathParameter(email)
+            .generateBaseUrl(baseUrl));
+    return VerifyEmailResponse.fromJson(result);
+  }
+
+  Future<UserSaaSResponse> signUpForSaaS(Uri baseUrl, SignUpBodyRequest signUpBodyRequest) async {
+    final result = await _dioClient.post(
+        Endpoint.signUp.generateBaseUrl(baseUrl),
+        data: signUpBodyRequest.toJson());
+    return UserSaaSResponse.fromJson(result);
+  }
+
```

Comment:
How about separate three of it to SaaSHttpClient?

Reply:
Please see [https://github.com/linagora/linshare-mobile-flutter-app/pull/759/commits/5caa903973270a1cf29dbecfb577794314d5d7ae ](https://github.com/linagora/linshare-mobile-flutter-app/pull/759/commits/5caa903973270a1cf29dbecfb577794314d5d7ae )